projects/client/RabbitMQ.Client/src/client/impl/TcpClientAdapter.cs

```diff
@@ -27,10 +27,17 @@ public TcpClientAdapter(Socket socket)
             AssertSocket();
             var adds = await Dns.GetHostAddressesAsync(host).ConfigureAwait(false);
             var ep = TcpClientAdapterHelper.GetMatchingHost(adds, sock.AddressFamily);
-            if (ep == default(IPAddress))
+
+            if (ep == null)
             {
+                if (adds.Length != 0)
+                {
+                    throw new NotSupportedException("Resolved ip address doesn't support " + sock.AddressFamily);
```

Comment:
> then these type of exception will be cough in the caller method and we will try to connect with IPv4 after.

I don't see where this exception is caught. Could you please point that out?

Reply:
SocketFrameHandler.ConnectOrFail calls TCPClientAdapter.ConnectAsync and catch several exception types including NotSupportedException.